camel/models/cohere_model.py

```diff
@@ -0,0 +1,240 @@
+            )
+        return self._token_counter
+
+    @api_keys_required("COHERE_API_KEY")
+    def run(self, messages: List[OpenAIMessage]) -> ChatCompletion:
+        """Runs inference of Cohere chat completion.
+
+        Args:
+            messages (List[OpenAIMessage]): Message list with the chat history
+                in OpenAI API format.
+
+        Returns:
+            ChatCompletion.
+        """
+        from cohere.core.api_error import ApiError
+
+        cohere_messages = self._to_cohere_chatmessage(messages)
+
+        # Filter out unsupported parameters
+        supported_params = {
+            'temperature',
+            'p',
+            'k',
+            'max_tokens',
+            'prompt_truncation',
+        }
+        filtered_config = {
+            k: v
+            for k, v in self.model_config_dict.items()
+            if k in supported_params
+        }
+
+        try:
+            response = self._client.chat(
+                message=cohere_messages[-1]["message"],
+                chat_history=cohere_messages[:-1],  # type: ignore[arg-type]
+                model=self.model_type.value,
+                **filtered_config,
+            )
+        except ApiError as e:
+            logging.error(f"Cohere API Error: {e.status_code}")
+            logging.error(f"Error body: {e.body}")
+            raise
+        except Exception as e:
+            logging.error(f"Unexpected error when calling Cohere API: {e!s}")
+            raise
+
+        openai_response = self._to_openai_response(response)
+
+        # Add AgentOps LLM Event tracking
+        if LLMEvent:
+            llm_event = LLMEvent(
+                thread_id=openai_response.id,
+                prompt=" ".join(
+                    [message.get("content") for message in messages]  # type: ignore[misc]
+                ),
+                prompt_tokens=openai_response.usage.prompt_tokens,  # type: ignore[union-attr]
+                completion=openai_response.choices[0].message.content,
+                completion_tokens=openai_response.usage.completion_tokens,  # type: ignore[union-attr]
+                model=self.model_type.value,
+            )
+            record(llm_event)
+
+        return openai_response
+
+    def check_model_config(self):
+        """Check whether the model configuration contains any
+        unexpected arguments to Cohere API.
+
+        Raises:
+            ValueError: If the model configuration dictionary contains any
+                unexpected arguments to Cohere API.
+        """
+        supported_params = {
+            'temperature',
+            'p',
+            'k',
+            'max_tokens',
+            'prompt_truncation',
+        }
+        for param in self.model_config_dict:
+            if param not in supported_params:
+                raise ValueError(
+                    f"Unexpected argument `{param}` is "
+                    "input into Cohere model backend."
+                )
```

Comment:
use COHERE_API_PARAMS in `camel/configs/cohere_config.py` to do the check

Reply:
Hi Wendong, I need to confirm if the function check_model_config should be implemented in camel/configs/cohere_config.py to perform the check, 
as I noticed that the checks for other models are implemented in camel/models/*_model.py.